Fixing install.sh so that the BASIC_ONLY option does not always skip the haskell installation.

tdfirth opened the pull request:
Problem was that the 'boolean' values used before were actually just an empty variable and a 'y' character. When the BASIC_ONLY parameter was passed into main, if --basic was not used then the variable was empty and so wasn't recognised as a parameter at all.

This meant that the 3rd parameter (the hoogle db one), got passed in as the second and as it was set to 'y' by default the Haskell binary installation was skipped.

I just replaced these pseudo booleans with actual booleans and it is a little bit simpler now as a result.  Then renamed the variables so that the logic made a little more sense.

jship commented on install.sh:
```diff
@@ -117,7 +118,7 @@ do_setup() {
   setup_vim $HVN_DEST
 
-  if test -z "$BASIC_ONLY"
+  if "$FULL_INSTALL"
```
Doing a string comparison here instead of `true`/`false` commands would be clearer.  See this link: http://stackoverflow.com/a/2953673

tdfirth replied:
Ah that's interesting, thanks for the feedback.  Have made that change and updated the one in setup.hs as well.  Let me know if there's something else.

begriffs commented:
Thanks for the PR, I'll check it out tonight.

begriffs commented:
Tried restarting the test on CI. It's not able to finish in under the time limit without its build cache. Maybe we should add `--dry-run` to the stack installs on CI. That way it will fail if it can't determine a build plan, but won't take forever trying to build things from scratch. Would you like to try adding that feature to this PR?

tdfirth commented:
Yeah no problem, should be able to do it later on today or maybe tomorrow.

begriffs commented:
This pull request is a big improvement because the master branch terminates prematurely.

However when I try this branch on a fresh machine I do get an error. After the solver finds that LTS-7.24 works, then it attempts to build and says this:

```
Using resolver: lts-7.24
Solver requires that cabal be on your PATH
Try running 'stack install cabal-install'

*** Solving dependencies for helper binaries failed with error 1.
*** Aborting...
```

tdfirth commented:
Hey, really sorry I took so long on this!  Things a bit busy at work right now.  I've added the dry run feature in and have tested it - it seems to be working well.  It still takes a fair while to execute but should be more than fast enough.  Have also updated .travis.yml.

Re your question on cabal-install...stack settling on lts-7.24 for me too but I don't seem to have that particular issue.  I don't even have cabal-install installed on the machine I'm using ('which cabal-install' returns 'cabal-install not found')...

begriffs commented on scripts/setup.sh:
```diff
@@ -85,11 +91,21 @@ setup_haskell() {
-    [ ${RETCODE} -ne 0 ] && exit_err "Installing helper binary dependency $dep failed with error ${RETCODE}."
-  done
+
+  if [ "$DRY_RUN" == false ]
```
Is it possible to move the conditional inside the loop to avoid duplicating the loop?

tdfirth replied:
Done, let me know if you need something else.

begriffs commented:
Thank you so much! This fixes a bad problem, and the dry-run on CI helps a lot too.

tdfirth commented:
Glad to have helped, my thanks to you for maintaining the greatest vim configuration on the planet.

jship commented:
Whoops, looks like I introduced this via #238.  Glad it's fixed.
